src/uct/ib/base/ib_md.c

```diff
@@ -1192,6 +1192,31 @@ static void uct_ib_md_check_dmabuf(uct_ib_md_t *md)
 #endif
 }
 
+static void uct_ib_md_check_odp(uct_ib_md_t *md)
+{
+    uct_md_mem_reg_params_t params = {};
+    const static size_t buffer_size = 8;
+    char buffer[buffer_size];
+    struct ibv_mr *mr;
+    ucs_status_t status;
+
+    if (IBV_ACCESS_ON_DEMAND == 0) {
+        ucs_debug("IBV_ACCESS_ON_DEMAND is not supported");
+        return;
+    }
+
+    status = uct_ib_reg_mr(md, buffer, buffer_size, &params,
+                           IBV_ACCESS_ON_DEMAND, NULL, &mr);
+    if (status != UCS_OK) {
+        ucs_debug("%s: ODP is not supported", uct_ib_device_name(&md->dev));
```

Comment:
does it really fail if ODP is not supported or just silently ignored?

Reply:
In that case I just copied behavior from `uct_ib_mlx5_devx_check_odp` so ODP is not supported it will print debug message and leave `reg_nonblock_mem_types` empty.